analyzer.go

```diff
@@ -15,41 +16,186 @@ type Platform interface {
 	API() *api.Version
 }
 
+type AnalyzerFactory struct {
+	platformAPI     *api.Version
+	cacheHandler    CacheHandler
+	configHandler   ConfigHandler
+	imageHandler    ImageHandler
+	registryHandler RegistryHandler
+}
+
+func NewAnalyzerFactory(platformAPI *api.Version, cacheHandler CacheHandler, configHandler ConfigHandler, imageHandler ImageHandler, registryHandler RegistryHandler) *AnalyzerFactory {
```

Comment:
It seemed like the constructor for a `lifecycle.Analyzer` belonged in the `lifecycle` package.